src/client/index.ts

```diff
@@ -657,3 +745,31 @@ export function btreeItemToAggregateItem<K extends Key, ID extends string>({
     sumValue: s,
   };
 }
+
+export type NamespacedArgs<Args, Namespace> =
+  Args & { namespace: Namespace } | (Namespace extends undefined ? Args : never);
+export type NamespacedOpts<Opts, Namespace> =
+  [{ namespace: Namespace } & Opts] | (
+    Namespace extends undefined
+      ? [] | [Opts]
```

Comment:
This might be possible to do with `[Opts?]`

Reply:
TIL that syntax